### PR TITLE
Pass readonly option on root mount

### PR DESCRIPTION
**Pull request by jiacheliu3**
This resolves issue https://github.com/Alluxio/alluxio/issues/10812#event-3012406507

Correctly propagate root mount options `alluxio.master.mount.table.root.readonly` to MountTable

After the change:
```
$ bin/alluxio fs mount                                                                                     
/xxx/alluxio/underFSStorage  on  /  (local, capacity=420.03GB, used=-1B(0%), read-only, not shared, properties={})
$ bin/alluxio getConf --master | grep readonly
alluxio.master.mount.table.root.readonly=true
alluxio.underfs.object.store.mount.shared.publicly=false
$ bin/alluxio fs copyFromLocal ./LICENSE /LICENSE                                                          
A write operation on /LICENSE under a readonly mount point / is not allowed
```

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -456,13 +456,22 @@ public DefaultFileSystemMaster(BlockMaster blockMaster, CoreMasterContext master
...
+        shared = ServerConfiguration.getBoolean(
+                PropertyKey.UNDERFS_OBJECT_STORE_MOUNT_SHARED_PUBLICLY);
+      } else {
+        shared = ServerConfiguration.getBoolean(
+                PropertyKey.MASTER_MOUNT_TABLE_ROOT_SHARED);
+      }
+      boolean readonly = ServerConfiguration.getBoolean(
+              PropertyKey.MASTER_MOUNT_TABLE_ROOT_READONLY);
       String rootUfsUri = ServerConfiguration.get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS);
-      boolean shared = resource.get().isObjectStorage()
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: @gpang Does the original implementation look like a bug to you?

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 2]
```diff
@@ -456,13 +456,22 @@ public DefaultFileSystemMaster(BlockMaster blockMaster, CoreMasterContext master
...
   private static MountInfo getRootMountInfo(MasterUfsManager ufsManager) {
     try (CloseableResource<UnderFileSystem> resource = ufsManager.getRoot().acquireUfsResource()) {
+      boolean shared;
+      if (resource.get().isObjectStorage()) {
+        shared = ServerConfiguration.getBoolean(
+                PropertyKey.UNDERFS_OBJECT_STORE_MOUNT_SHARED_PUBLICLY);
+      } else {
+        shared = ServerConfiguration.getBoolean(
+                PropertyKey.MASTER_MOUNT_TABLE_ROOT_SHARED);
+      }
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: I tried to update `shared` to this logic which seems more correct. And subsequently found issue https://github.com/Alluxio/alluxio/issues/10896
The issue related to `shared` seems deeper than I thought. I will revert the change regarding `shared` field and focus on `readonly` in this PR. I will attempt on the `shared` field in a separate PR later.

**Comment by jiacheliu3** [comment]
@gpang I tried to add a test but wasn't able to.

There's no existing test for `DefaultFileSystemMaster` and I think the workload creating one deserves a separate PR already...

And I don't think the unit/integration tests for either MountCommand or ReportCommand are good places to test this.

Do you have a better idea where the tests can be?

**Comment by jiacheliu3** [comment]
@gpang PTAL there's one integration test added on readonly root mount. Thanks!

File: tests/src/test/java/alluxio/client/fs/FileSystemReadonlyTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Map;
+
+/**
+ * Test behavior of {@link FileSystemMaster}, when the paths are readonly.
+ */
+public class FileSystemReadonlyTest {
+  @Rule
```
Comment (gpang) [inline tests/src/test/java/alluxio/client/fs/FileSystemReadonlyTest.java]: Can we make this a `ClassRule` so we dont have to re-create the cluster for each test? Starting a cluster is costly. (I think this also means the variable will have to be static)

Comment (jiacheliu3) [reply tests/src/test/java/alluxio/client/fs/FileSystemReadonlyTest.java]: Done. BTW most integration tests are still using `@Rule`. We might want to batch-fix that in the future.

Comment (gpang) [reply tests/src/test/java/alluxio/client/fs/FileSystemReadonlyTest.java]: Yes, I know. That is why I'm trying to make new usages use classrule. Thanks!

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**